lib/std/typedthreads.nim

```diff
@@ -33,6 +43,34 @@
 ##
 ##   deinitLock(L)
 ##   ```
+## 
+## You can pass pointer to threads, but the memory must outlive the thread.
+## 
+## ```Nim
+## import locks
+## 
+## var l: Lock
+## 
+## proc threadFunc(obj: ptr seq[int]) {.thread.} =
+##     withLock l:
+##         for i in 0..<100:
+##             obj[].add(obj[].len * obj[].len)
```

Comment:
That is only supported for `mm:arc`, `mm:boehm`. `mm:orc` and `mm:atomicArc` so either document this fact in bold letters or choose a different example.

Reply:
Added **not valid** in bold, it should catch the eye of somebody skimming though the documentation.